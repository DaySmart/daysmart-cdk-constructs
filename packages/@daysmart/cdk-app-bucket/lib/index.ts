import cdk = require('@aws-cdk/core');
import s3 = require('@aws-cdk/aws-s3');

export class AppBucket extends cdk.Construct {
    constructor(scope: cdk.Construct, id: string, props: any) {
        super(scope, id);

        new s3.Bucket(this, 'Bucket', {
            bucketName: `${props.stage}-${props.appName.toLowerCase()}-website-bucket`,
            publicReadAccess: true,
            versioned: true,
        });
    }
}