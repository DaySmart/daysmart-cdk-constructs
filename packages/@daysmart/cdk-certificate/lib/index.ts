import * as cdk from '@aws-cdk/core';
import acm = require('@aws-cdk/aws-certificatemanager');
import route53 = require('@aws-cdk/aws-route53');

export interface CdkCertificateProps {
    domainName: string;
    stage: string;
    project: string;
}

export class CdkCertificate extends cdk.Construct {

  constructor(scope: cdk.Construct, id: string, props: CdkCertificateProps) {
    super(scope, id);

    const hostedZone = route53.HostedZone.fromLookup(this, 'HostedZone', {
        domainName: props.domainName
    });

    new acm.Certificate(this, 'Certificate', {
        domainName: props.domainName,
        subjectAlternativeNames: [
            `${props.stage}.${props.project}.${props.domainName}`,
            `*.${props.stage}.${props.project}.${props.domainName}`
        ],
        validation: acm.CertificateValidation.fromDns(hostedZone)
    });
  }
}
