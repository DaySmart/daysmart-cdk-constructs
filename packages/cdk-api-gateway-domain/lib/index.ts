import * as cdk from '@aws-cdk/core';
import * as apigw from '@aws-cdk/aws-apigateway';
import * as acm from '@aws-cdk/aws-certificatemanager';
import * as route53 from '@aws-cdk/aws-route53';

export interface CdkApiGatewayDomainProps {
  companyDomainName: string;
  companyHostedZoneId: string;
  project: string;
  baseEnv: string;
  dynamicEnv?: string;
  certificateArn: string;
  restApiId: string;
  basePath: string;
}

export class CdkApiGatewayDomain extends cdk.Construct {

  constructor(scope: cdk.Construct, id: string, props: CdkApiGatewayDomainProps) {
    super(scope, id);

    const api = apigw.RestApi.fromRestApiId(this, "Rest API", `${props.restApiId}`);

    let customDomain: apigw.DomainName;

    if (props.dynamicEnv) {
      customDomain = new apigw.DomainName(this, 'Custom Domain', {
        domainName: `${props.dynamicEnv}-api.${props.baseEnv}.${props.project}.${props.companyDomainName}`,
        certificate: acm.Certificate.fromCertificateArn(this, "Certificate", `${props.certificateArn}`),
        endpointType: apigw.EndpointType.EDGE,
        securityPolicy: apigw.SecurityPolicy.TLS_1_0
      });
    } else {
      customDomain = new apigw.DomainName(this, 'Custom Domain', {
        domainName: (props.baseEnv == "prod") ? `api.${props.project}.${props.companyDomainName}` : `api.${props.baseEnv}.${props.project}.${props.companyDomainName}`,
        certificate: acm.Certificate.fromCertificateArn(this, "Certificate", `${props.certificateArn}`),
        endpointType: apigw.EndpointType.EDGE,
        securityPolicy: apigw.SecurityPolicy.TLS_1_0
      });
    }

    const cloudformationBasePathMapping = new apigw.CfnBasePathMapping(this, "CloudformationBasePathMapping", {
      basePath: `${props.basePath}`,
      domainName: customDomain.domainName,
      restApiId: `${props.restApiId}`,
      stage: `${props.baseEnv}` 
    });

    const cloudformationRoute53Change = new route53.CfnRecordSet(this, "CloudformationAliasRecordRoute53", {
      aliasTarget: {
        dnsName: customDomain.domainNameAliasDomainName,
        evaluateTargetHealth: false,
        hostedZoneId: "Z2FDTNDATAQYW2"
      },
      comment: `CloudFront distribution for ${customDomain.domainName}`,
      hostedZoneId: props.companyHostedZoneId,
      name: customDomain.domainName,
      type: "A"
    });
  }
}
