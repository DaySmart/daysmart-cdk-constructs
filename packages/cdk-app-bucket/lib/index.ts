import cdk = require('@aws-cdk/core');
import s3 = require('@aws-cdk/aws-s3');
import cloudfront = require('@aws-cdk/aws-cloudfront');
import iam = require('@aws-cdk/aws-iam');

export interface AppBucketProps {
    stage: string;
    appName: string;
    dynamicEnvName: string;
    projectName: string;
    sharedServicesAccountId?: string;
}
export class AppBucket extends cdk.Construct {
    constructor(scope: cdk.Construct, id: string, props: AppBucketProps) {
        super(scope, id);

        const bucket = new s3.Bucket(this, 'Bucket', {
            bucketName: `${props.dynamicEnvName}-${props.appName}.${props.stage}.${props.projectName}`,
            versioned: true,
            objectOwnership: s3.ObjectOwnership.BUCKET_OWNER_PREFERRED,
            blockPublicAccess: s3.BlockPublicAccess.BLOCK_ALL
        });

        const originAccessIdentity = new cloudfront.OriginAccessIdentity(this, 'OriginAcessIdentity', {
            comment: `OriginAccessIdentity for ${bucket.bucketName}.`
        });

        bucket.grantRead(originAccessIdentity)

        if(props.sharedServicesAccountId) {
            bucket.grantReadWrite(new iam.AccountPrincipal(props.sharedServicesAccountId));
        }

        let output = new cdk.CfnOutput(this, "AppBucket", {
            value: bucket.bucketName
        });

        let output2 = new cdk.CfnOutput(this, "OriginAccessIdentity", {
            value: originAccessIdentity.originAccessIdentityName
        });

        output.overrideLogicalId("AppBucket");
        output2.overrideLogicalId("OriginAccessIdentity");
    }
}