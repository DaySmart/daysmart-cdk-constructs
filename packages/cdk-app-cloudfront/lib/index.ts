import * as cdk from 'aws-cdk-lib';
import { Construct } from 'constructs';
import * as s3 from 'aws-cdk-lib/aws-s3';
import * as cloudfront from 'aws-cdk-lib/aws-cloudfront'
import * as route53 from 'aws-cdk-lib/aws-route53'
import * as targets from 'aws-cdk-lib/aws-route53-targets'

export interface CdkAppCloudfrontProps {
    s3BucketName: string;
    originAccessIdentity: string;
    loggingBucketName?: string;
    stage: string;
    project: string;
    certArn: string;
    companyDomainName: string;
    companyHostedZoneId: string;
    dynamicEnvName?: string;
    appName?: string;
    projectHostedZoneId?: string;
    projectDomainName?: string;
}

export class CdkAppCloudfront extends Construct {

  constructor(scope: Construct, id: string, props: CdkAppCloudfrontProps) {
    super(scope, id);

    const codeBucket = s3.Bucket.fromBucketName(
        this,
        'CodeBucket',
        props.s3BucketName
    );

    const originAccessIdentity = cloudfront.OriginAccessIdentity.fromOriginAccessIdentityName(
        this,
        "OriginAccessIdentity",
        props.originAccessIdentity
    );

    let aliases: string[] = [];
    aliases.push(...this.getCompanyDomainAliases(props));

    if(props.projectDomainName && props.projectHostedZoneId) {
        aliases.push(...this.getProjectDomainAliases(props));
    }

    const distribution = new cloudfront.CloudFrontWebDistribution(this, 'Distribution', {
        originConfigs: [
            {
                s3OriginSource: {
                    s3BucketSource: codeBucket,
                    originAccessIdentity: originAccessIdentity
                },
                behaviors: [
                    {
                        isDefaultBehavior: true,
                        allowedMethods: cloudfront.CloudFrontAllowedMethods.ALL,
                        forwardedValues: {
                            queryString: true,
                            cookies: {
                              forward: "all",
                            },
                            headers: ["Referer"],
                        },
                    }
                ]
            }
        ],
        viewerProtocolPolicy: cloudfront.ViewerProtocolPolicy.REDIRECT_TO_HTTPS,
        priceClass: cloudfront.PriceClass.PRICE_CLASS_ALL,
        httpVersion: cloudfront.HttpVersion.HTTP2,
        loggingConfig: props.loggingBucketName ? {
            bucket: s3.Bucket.fromBucketName(
                this,
                "CloudfrontLoggingBucket",
                props.loggingBucketName
            ),
            prefix: `${props.stage}-${props.project}`,
            includeCookies: true
        } : undefined,
        errorConfigurations: [
            {
              errorCode: 404,
              responseCode: 200,
              errorCachingMinTtl: 1,
              responsePagePath: "/index.html",
            },
            {
              errorCode: 403,
              responseCode: 200,
              errorCachingMinTtl: 1,
              responsePagePath: "/index.html",
            },
        ],
        comment: `${props.stage} ${props.project}`,
        viewerCertificate: {
          props: {
            acmCertificateArn: props.certArn,
            sslSupportMethod: "sni-only",
            minimumProtocolVersion: "TLSv1.1_2016",
          },
          aliases: aliases
        },
    });

    const cloudfrontTarget = new targets.CloudFrontTarget(distribution);

    const companyHostedZone = route53.HostedZone.fromHostedZoneAttributes(this, 'HostedZone', {
        hostedZoneId: props.companyHostedZoneId,
        zoneName: props.companyDomainName
    });

    this.getCompanyDomainAliases(props).forEach(alias => {
        new route53.ARecord(this, alias, {
            zone: companyHostedZone,
            target: route53.RecordTarget.fromAlias(cloudfrontTarget),
            recordName: alias
        });
    });

    if(props.projectDomainName && props.projectHostedZoneId) {
        const projectHostedZone = route53.HostedZone.fromHostedZoneAttributes(this, 'ProjectHostedZone', {
            hostedZoneId: props.projectHostedZoneId,
            zoneName: props.projectDomainName
        });

        this.getProjectDomainAliases(props).forEach(alias => {
            new route53.ARecord(this, alias, {
                zone: projectHostedZone,
                target: route53.RecordTarget.fromAlias(cloudfrontTarget),
                recordName: alias
            });
        });
    }

    let output = new cdk.CfnOutput(this, "CloudfrontDistribution", {
        value: distribution.distributionId
    });

    let output2 = new cdk.CfnOutput(this, "DistributionDomainName", {
        value: distribution.distributionDomainName
    })

    output.overrideLogicalId("CloudfrontDistribution");
    output2.overrideLogicalId("DistributionDomainName");
  }

  getCompanyDomainAliases(props: CdkAppCloudfrontProps): Array<string> {
    let aliases: Array<string> = [];
    if(props.dynamicEnvName && props.appName) {
        aliases.push(`${props.dynamicEnvName}-${props.appName}.${props.stage}.${props.project}.${props.companyDomainName}`);
    } else if(props.appName) {
        aliases.push(`${props.appName}.${props.stage}.${props.project}.${props.companyDomainName}`);
    } else {
        aliases.push(`${props.stage}.${props.project}.${props.companyDomainName}`);

        if(props.stage === 'prod') {
            aliases.push(`${props.project}.${props.companyDomainName}`);
        }
    }

    return aliases;
  }

  getProjectDomainAliases(props: CdkAppCloudfrontProps): Array<string> {
    let aliases: Array<string> = []
    if(props.dynamicEnvName && props.appName) {
        aliases.push(`${props.dynamicEnvName}-${props.appName}.${props.stage}.${props.projectDomainName}`);
    } else if (props.appName) {
        aliases.push(`${props.appName}.${props.stage}.${props.projectDomainName}`);
    } else {
        aliases.push(`${props.stage}.${props.projectDomainName}`);
    }

    if(props.stage === "prod") {
        if(props.appName) {
            aliases.push(`${props.appName}.${props.projectDomainName}`);
        } else {
            aliases.push(props.projectDomainName as string);
        }
    }
    

    return aliases;
  }
}
