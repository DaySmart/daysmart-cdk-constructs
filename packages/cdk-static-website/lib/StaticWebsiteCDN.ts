import * as cdk from '@aws-cdk/core';
import * as acm from '@aws-cdk/aws-certificatemanager';
import * as cloudfront from '@aws-cdk/aws-cloudfront';
import * as origins from '@aws-cdk/aws-cloudfront-origins';
import * as s3 from '@aws-cdk/aws-s3';
import * as route53 from '@aws-cdk/aws-route53';
import * as alias from '@aws-cdk/aws-route53-targets';


/**
 * Static website CDN properties
 */
export interface StaticWebsiteCDNProps {
    /**
     * The name of the S3 bucket hosting the static website
     */
    bucketName: string;

    /**
     * The Origin Access Identity for the S3 bucket required to read
     */
    originAccessIdentity: string;

    /**
     * The ACM certificate ARN for the domain names
     */
    certificateArn: string;

    /**
     * A list of domain names that will be aliases on the Cloudfront distribution
     */
    domainNames: string[];

    /**
     * The name of the hosted zone that includes all of the domain names given
     */
    hostedZoneDomain: string;

    /**
     * A custom origin path in the S3 bucket for the CloudFront origin
     */
    originPath?: string;
}

/**
 * A CDN for a static website that includes a cloudfront distribution that directs traffic to
 * an S3 bucket and includes a Route 53 record set
 */
export class StaticWebsiteCDN extends cdk.Construct {
    constructor(scope: cdk.Construct, id: string, props: StaticWebsiteCDNProps) {
        super(scope, id);

        const appBucket = s3.Bucket.fromBucketName(this, 'AppBucket', props.bucketName);

        const distribution = new cloudfront.Distribution(this, 'Distribution', {
            defaultBehavior: {
                origin: new origins.S3Origin(appBucket, {
                    originAccessIdentity: cloudfront.OriginAccessIdentity.fromOriginAccessIdentityName(this, 'OriginAccessIdentity', props.originAccessIdentity),
                    originPath: props.originPath
                }),
                viewerProtocolPolicy: cloudfront.ViewerProtocolPolicy.REDIRECT_TO_HTTPS
            },
            certificate: acm.Certificate.fromCertificateArn(this, 'Certificate', props.certificateArn),
            domainNames: props.domainNames,
            enableLogging: true,
            logIncludesCookies: true,
            defaultRootObject: 'index.html',
            errorResponses: [
                {
                    httpStatus: 404,
                    responseHttpStatus: 200,
                    responsePagePath: '/index.html'
                }
            ]
        })

        const hostedZone = route53.HostedZone.fromLookup(this, 'HostedZone', {
            domainName: props.hostedZoneDomain
        });

        props.domainNames.forEach((domainName, i) => {
            new route53.ARecord(this, `Alias${i}`, {
                zone: hostedZone,
                recordName: domainName,
                target: route53.RecordTarget.fromAlias(new alias.CloudFrontTarget(distribution))
            }) 
        })
        const distributionIdOutput = new cdk.CfnOutput(this, 'DistributionId', {
            value: distribution.distributionId
        });
        distributionIdOutput.overrideLogicalId('DistributionId');

        const distributionDomainNameOutput = new cdk.CfnOutput(this, 'DistributionDomainName', {
            value: distribution.distributionDomainName
        });
        distributionDomainNameOutput.overrideLogicalId('DistributionDomainName');
    }
}