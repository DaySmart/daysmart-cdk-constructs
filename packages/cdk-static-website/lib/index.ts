export { AppBucket, AppBucketProps } from './app-bucket';
export { StaticWebsiteCDN, StaticWebsiteCDNProps } from './StaticWebsiteCDN';
export { CdkS3Deployment, CdkS3DeploymentProps } from './s3-deployment';