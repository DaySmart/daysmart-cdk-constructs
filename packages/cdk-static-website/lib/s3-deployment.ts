import * as path from 'path'; 
import * as cdk from 'aws-cdk-lib';
import * as lambda from 'aws-cdk-lib/aws-lambda';
import { Source, SourceConfig, ISource } from 'aws-cdk-lib/aws-s3-deployment';
import * as s3 from 'aws-cdk-lib/aws-s3';
import * as cloudfront from 'aws-cdk-lib/aws-cloudfront';
import * as iam from 'aws-cdk-lib/aws-iam';
import { AwsCliLayer } from 'aws-cdk-lib/lambda-layer-awscli';
import { Construct } from 'constructs';

export interface CdkS3DeploymentProps {
  bucketName: string;
  distributionId: string;
  distributionDomain: string;
  sourceDir?: string;
  distributionPath?: string;
  invalidationPath?: string;
  environment: string;
  snsTopicArn: string;
}

export class CdkS3Deployment extends Construct {

  constructor(scope: Construct, id: string, props: CdkS3DeploymentProps) {
    super(scope, id);

    const bucket = s3.Bucket.fromBucketName(this, 'Bucket', props.bucketName);

    const distribution = cloudfront.Distribution.fromDistributionAttributes(this, 'Distribution', {
      distributionId: props.distributionId,
      domainName: props.distributionDomain
    });

    let destinationPrefix = props.distributionPath;
    if(destinationPrefix && destinationPrefix.startsWith('/')) {
      destinationPrefix = destinationPrefix.substring(1);
    }

    const handler = new lambda.SingletonFunction(this, 'BucketDeploymentHandler', {
      code: lambda.Code.fromAsset(path.join(__dirname, 'lambda')),
      layers: [new AwsCliLayer(this, 'AWSCliLayer')],
      runtime: lambda.Runtime.PYTHON_3_9,
      handler: 'handler.handler',
      lambdaPurpose: 'Custom::BucketDeployment',
      timeout: cdk.Duration.minutes(15),
      uuid: '8693BB64-9689-44B6-9AAF-B0CC9EB8756C'
    });

    bucket.grantReadWrite(handler);

    const handlerRole = handler.role;
    if(!handlerRole) { throw new Error('lambda.SingletonFunction should have created a Role'); }
    const sources: SourceConfig[] = [Source.asset(props.sourceDir ? props.sourceDir : '../dist')].map((source: ISource) => source.bind(this, {  handlerRole }));

    handler.addToRolePolicy(new iam.PolicyStatement({
      effect: iam.Effect.ALLOW,
      actions: [
        'cloudfront:GetInvalidation',
        'cloudfront:CreateInvalidation',
        'cloudfront:ListDistributions'
      ],
      resources: ['*']
    }));

    handler.addToRolePolicy(new iam.PolicyStatement({
      effect: iam.Effect.ALLOW,
      actions: [
        'sns:Publish'
      ],
      resources: [`${props.snsTopicArn}`]
    }));

    new cdk.CustomResource(this, 'CustomResource', {
      serviceToken: handler.functionArn,
      resourceType: 'Custom::BucketDeployment',
      properties: {
        SourceBucketNames: sources.map(source => source.bucket.bucketName),
        SourceObjectKeys: sources.map(source => source.zipObjectKey),
        DestinationBucketName: bucket.bucketName,
        DestinationBucketKeyPrefix: destinationPrefix,
        DistributionId: distribution.distributionId,
        DistributionPaths: [props.invalidationPath ? props.invalidationPath : '/*'],
        Environment: props.environment,
        SnsTopicArn: props.snsTopicArn
      }
    });
  }
}
