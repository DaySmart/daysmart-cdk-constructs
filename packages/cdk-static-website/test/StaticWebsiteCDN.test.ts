import * as cdk from '@aws-cdk/core';
import '@aws-cdk/assert/jest'
import { SynthUtils } from '@aws-cdk/assert';
import { StaticWebsiteCDN } from '../lib';

test('Static Website CDN template', () => {
    const stack = new cdk.Stack(undefined, 'stack', {
        env: {
            account: '123456',
            region: 'us-east-1'
        }
    });
    
    new StaticWebsiteCDN(stack, 'WebsiteCDN', {
        bucketName: 'my-bucket',
        certificateArn: 'arn:aws:acm:us-east-1:123456:certificate/blah',
        domainNames: ['example.domain.com'],
        hostedZoneDomain: 'example.domain.com',
        originAccessIdentity: 'oai-123'
    })
    expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
});