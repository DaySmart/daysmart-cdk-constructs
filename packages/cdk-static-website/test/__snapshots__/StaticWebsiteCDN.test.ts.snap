// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Static Website CDN template 1`] = `
Object {
  "Mappings": Object {
    "AWSCloudFrontPartitionHostedZoneIdMap": Object {
      "aws": Object {
        "zoneId": "Z2FDTNDATAQYW2",
      },
      "aws-cn": Object {
        "zoneId": "Z3RFFRIM2A3IF5",
      },
    },
  },
  "Resources": Object {
    "WebsiteCDNAlias0ECFCB318": Object {
      "Properties": Object {
        "AliasTarget": Object {
          "DNSName": Object {
            "Fn::GetAtt": Array [
              "WebsiteCDNDistribution87C8E1D8",
              "DomainName",
            ],
          },
          "HostedZoneId": Object {
            "Fn::FindInMap": Array [
              "AWSCloudFrontPartitionHostedZoneIdMap",
              Object {
                "Ref": "AWS::Partition",
              },
              "zoneId",
            ],
          },
        },
        "HostedZoneId": "DUMMY",
        "Name": "example.domain.com.",
        "Type": "A",
      },
      "Type": "AWS::Route53::RecordSet",
    },
    "WebsiteCDNDistribution87C8E1D8": Object {
      "Properties": Object {
        "DistributionConfig": Object {
          "Aliases": Array [
            "example.domain.com",
          ],
          "CustomErrorResponses": Array [
            Object {
              "ErrorCode": 404,
              "ResponseCode": 200,
              "ResponsePagePath": "/index.html",
            },
          ],
          "DefaultCacheBehavior": Object {
            "CachePolicyId": "658327ea-f89d-4fab-a63d-7e88639e58f6",
            "Compress": true,
            "TargetOriginId": "stackWebsiteCDNDistributionOrigin1BCB5126C",
            "ViewerProtocolPolicy": "redirect-to-https",
          },
          "DefaultRootObject": "index.html",
          "Enabled": true,
          "HttpVersion": "http2",
          "IPV6Enabled": true,
          "Logging": Object {
            "Bucket": Object {
              "Fn::GetAtt": Array [
                "WebsiteCDNDistributionLoggingBucketA7CD51F4",
                "RegionalDomainName",
              ],
            },
            "IncludeCookies": true,
          },
          "Origins": Array [
            Object {
              "DomainName": Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    "my-bucket.s3.us-east-1.",
                    Object {
                      "Ref": "AWS::URLSuffix",
                    },
                  ],
                ],
              },
              "Id": "stackWebsiteCDNDistributionOrigin1BCB5126C",
              "S3OriginConfig": Object {
                "OriginAccessIdentity": "origin-access-identity/cloudfront/oai-123",
              },
            },
          ],
          "ViewerCertificate": Object {
            "AcmCertificateArn": "arn:aws:acm:us-east-1:123456:certificate/blah",
            "MinimumProtocolVersion": "TLSv1.2_2019",
            "SslSupportMethod": "sni-only",
          },
        },
      },
      "Type": "AWS::CloudFront::Distribution",
    },
    "WebsiteCDNDistributionLoggingBucketA7CD51F4": Object {
      "DeletionPolicy": "Retain",
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Retain",
    },
  },
}
`;
